lib/go/thrift/simple_server.go

```diff
@@ -211,6 +211,10 @@ func (p *TSimpleServer) innerAccept() (int32, error) {
 			select {
 			case <-ctx.Done():
 				// client exited, do nothing
+				if client != nil {
+					client.Close()
```

Comment:
can you explain more why do we need to close the client here? is there any connection leak otherwise?

Reply:
Yes, if the call to Close is not called, then any dynamically allocated data in conn object is lost forever and hence the caller's golang runtime keeps eating memory.

Reply:
How would that affect overall performance, e.g. when doing multiple calls?

Reply:
Could you help clarify what is your overall performance concern? 

Below is my understanding of how the simpleServer works with its serverTransport regarding accept an incoming request. Please correct me if I miss anything.
A client object is created by calling the underlying serverTransport.Accept(). I think client.Close() is to signal serverTransport that *thrift server* has finished using this client. So serverTransport could do whatever is required on its part, either to cleanup or recycle the object for reuse. 

Without the signal from thrift server, what is expected from serverTransport to manage *client* object created by itself? The reference to client is lost after processing the request.

I guess thrift simple server has some assumption on how serverTransport to manage client created by serverTransport itself. Could you help advise where I could find the assumption?

Reply:
I simply asked a question. If your answer is a validated "No it will not affect perf because ..." that's fine with me.